sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java

```diff
@@ -373,24 +372,28 @@
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getPageRangesWithResponse#BlobRange-BlobAccessConditions}
+     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getPageRangesWithResponse#PageRange-BlobAccessConditions}
      *
-     * @param blobRange {@link BlobRange}
+     * @param pageRange {@link PageRange}
      * @param accessConditions {@link BlobAccessConditions}
      * @return A reactive response emitting all the page ranges.
      */
-    public Mono<Response<PageList>> getPageRangesWithResponse(BlobRange blobRange,
+    public Mono<Response<PageList>> getPageRangesWithResponse(PageRange pageRange,
         BlobAccessConditions accessConditions) {
-        return withContext(context -> getPageRangesWithResponse(blobRange, accessConditions, context));
+        return withContext(context -> getPageRangesWithResponse(pageRange, accessConditions, context));
     }
 
-    Mono<Response<PageList>> getPageRangesWithResponse(BlobRange blobRange, BlobAccessConditions accessConditions,
+    Mono<Response<PageList>> getPageRangesWithResponse(PageRange pageRange, BlobAccessConditions accessConditions,
         Context context) {
-        blobRange = blobRange == null ? new BlobRange(0) : blobRange;
+        if (pageRange == null) {
+            // Throwing is preferred to Single.error because this will error out immediately instead of waiting until
```

Comment:
Why are we throwing an error here now? Also, if we keep this let's not use RxJava terminology.